baseband/vlbi_base/base.py

```diff
@@ -70,6 +71,31 @@ def __init__(self, fh_raw, header0, nchan, bps, complex_data, thread_ids,
 
         self.frames_per_second = frames_per_second
         self.offset = 0
+        self.squeeze = squeeze
+
+    @property
+    def sample_shape(self):
+        """Shape of a data sample (possibly squeezed).
+        """
+        if self.squeeze:
+            if not self._squeezed_shape:
+                dims = np.array(self._sample_shape)
+                keep_dims = (dims > 1)
+                try:
+                    field_names = np.array(self._sample_shape._fields)
+                except AttributeError:
+                    self._squeezed_shape = tuple(dims[keep_dims])
+                else:
+                    sqz_shp = namedtuple('SampleShape', field_names[keep_dims])
+                    self._squeezed_shape = sqz_shp(*dims[keep_dims])
+            return self._squeezed_shape
+        return self._sample_shape
+
+    def _unsqueeze(self, data):
+        unity_dims = np.where(np.array(self._sample_shape) == 1)[0] + 1
```

Comment:
This can be done more directly:
```
new_shape = list(data.shape)
for i, dim in enumerate(self._sample_shape):
    if dim == 1:
        new_shape.insert(i, 1)  # maybe i+1?
return data.reshape(new_shape)
```
Actually: just to be sure, this loop may have to go backwards (in case there is more than one unity dimension). Do write some tests directly testing this logic!

Reply:
I agree that's more direct and doesn't require higher-level machinery (though I liked how compact mine was).  Changed, and included tests in `test_vlbi_base.py`.